packages/server/lib/api.coffee

```diff
@@ -34,6 +35,8 @@ if intervals = process.env.API_RETRY_INTERVALS
 
 rp = request.defaults (params = {}, callback) ->
   _.defaults(params, {
+    agent: agent
```

Comment:
Since we know the tests work through the agent - we need to add a single test to ensure that it's respecting the `agent` property here.

This is a critical piece of the entirety of the test runner - so much that if this one value was not present, users in corporate proxy environments would *not* ever be able to set their project up on our dashboard or log in.

You have real integration tests in other places that ensure the `agent` does the right thing - and you could add a single one to the `api_spec` that really stands up servers - but perhaps you could simply verify that the `agent` property is being passed here without having to spin up a whole world.

On the other hand - you've made it pretty damn easy to very the proxy is working correctly via your `DebugProxy` module, and it may be worth having a single "real" test because of the criticality of this.

As it stands, I believe that if you were to remove this property - all the tests would still pass, which means it's never once being "proven" that it's working.

Reply:
I added some tests that show that `agent.addRequest` is being called when API requests are being made. I think the proxy functionality is covered well enough elsewhere that this should be enough.